pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -213,6 +221,24 @@ func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconci
 
 		// apply remediation logic, if at least one condition last more than specified timeout
 		if unhealthyForTooLong(nodeCondition, conditionTimeout) {
+			if isBareMetalProvider(infra) {
+				// we already have reboot annotation on the node, stop reconcile
+				if _, ok := node.Annotations[annotationReboot]; ok {
+					return reconcile.Result{}, nil
```

Comment:
Does this mean it is up to the entity processing the annotation to power off the node delete the `Node` if reboot fails?

Reply:
exactly, I created two controllers under https://github.com/cynepco3hahue/machine-remediation-operator repository
- the first one is responsible to monitor nodes and create the `MachineRemediation` object in case if it recognizes the reboot annotation on the node(we can remove this one, ones the MHC will create the `MachineRemediation` object instead of add annotation on the node)
- the second one, monitors `MachineRemediation` and applies relevant remediation logic, it includes process rebooting, removing reboot annotation from the node once the reboot succeeded and remove the node once the reboot failed